pkg/telemetry/common.go

```diff
@@ -1,20 +1,37 @@
 package telemetry
 
 import (
+	"net/url"
+	"strings"
+
 	"github.com/pkg/errors"
 	"go.opentelemetry.io/otel"
 
-	"github.com/bacalhau-project/bacalhau/pkg/version"
 	"github.com/rs/zerolog/log"
 	"go.opentelemetry.io/otel/sdk/resource"
 	semconv "go.opentelemetry.io/otel/semconv/v1.17.0"
+
+	"github.com/bacalhau-project/bacalhau/pkg/analytics"
+	"github.com/bacalhau-project/bacalhau/pkg/version"
 )
 
 func SetupFromEnvs() {
 	newTraceProvider()
 	newMeterProvider()
 
 	otel.SetErrorHandler(otel.ErrorHandlerFunc(func(err error) {
+		switch e := err.(type) {
+		// this is the error returned by the analytics provider if the collector cannot be reached.
+		// We ignore it for this case as the error is benign.
+		case *url.Error:
+			if strings.Contains(e.URL, analytics.DefaultOtelCollectorEndpoint) {
+				return
+			} else {
+				log.Err(err).Msg("Error occurred while handling spans")
+			}
+		default:
+			// fall through and log the error
+		}
 		log.Err(err).Msg("Error occurred while handling spans")
```

Comment:
you are still logging the error at the end and will log the error twice for non analytics errors